tests/phpunit/test-emitter.php

```diff
@@ -623,4 +623,55 @@ public function test_surrogate_keys_with_filter_override() {
 		remove_all_filters("pantheon_should_add_terms");
 	}
 
+	/**
+	 * Test that queries on archive pages with multiple post types doesn't throw an error.
+	 *
+	 * @see https://getpantheon.atlassian.net/browse/BUGS-6934
+	 */
+	public function test_surrogate_keys_multiple_post_types() {
+		global $wp_query;
+
+		// Set up posts of different post types.
+		$this->factory->post->create( [
+			'post_type' => 'post',
+			'post_title' => 'test product'
+		] );
+		$this->factory->post->create( [
+			'post_type' => 'post',
+			'post_title' => 'test post'
+		] );
```

Comment:
these are not different post_types, was the first one supposed to be product?

Reply:
yep. Weird that the archive tests still passed. I'm adding some new tests to make sure both of these created post IDs show up in the surrogate keys.